soc/nxp/imx/imx7d/Kconfig.defconfig.mcimx7d_m4

```diff
@@ -11,4 +11,11 @@ config SYS_CLOCK_HW_CYCLES_PER_SEC
 config GPIO
 	default y
 
+#By default ROM_START is relocated to address 0 in a 1K region, allowing imx_rproc
+#to translate it to OCRAM_S address
+#Select the relocation if the chosen flash is not the OCRAM_S
+config ROMSTART_RELOCATION_ROM
+	default y if $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_FLASH))!=$(dt_nodelabel_reg_addr_hex,ocram_s_code)
```

Comment:
As the size of zephyr.bin may be enlarged, and romstart relocation is not necessary for some usecase even in case of startrom is not located at ocram_s,  so let's remove this enablement by dts checking, and enable CONFIG_ROMSTART_RELOCATION_ROM by manual, that is to say application decide whether enable it or not, so by default it is disabled.

But we can set default remstart relocation address just like:
config ROMSTART_REGION_ADDRESS
           default $(dt_nodelabel_reg_addr_hex,ocram_s_code)

Reply:
Yes the size of bin is enlarged, but the bin file can only be loaded with imx_fwloader or u-boot which are loaders that explicitly copy the boot-vector into OCRAM_S (imx7d).

I can remove the automatic enabling here if it's creating trouble. 

I will split the commits :)

Reply:
@JiafeiPan  I don't know why CI failed on the last push as local compliance check and checkpatch were okay.
Just rebased on main and pushed again to retrig the CI checks.